modules/tests/test/testconfigs/copy-template-config.go

```diff
@@ -25,7 +32,28 @@ type CopyTemplateTestConfig struct {
 }
 
 func (c *CopyTemplateTestConfig) Init(options *testoptions.TestOptions) {
+	c.TaskData.ResultTemplateName = NewTemplateName
+
+	if c.TaskData.SourceTemplateName == "" && !c.TaskData.DoNotSetSourceName {
+		c.TaskData.SourceTemplateName = testtemplate.CirrosTemplateName
+	}
+
+	if c.TaskData.TargetTemplateName == "" && !c.TaskData.DoNotSetTargetName {
+		c.TaskData.TargetTemplateName = NewTemplateName
+	}
+
 	c.deploymentNamespace = options.DeployNamespace
+
+	c.TaskData.SourceNamespace = options.ResolveNamespace(c.TaskData.SourceTemplateNamespace, options.TestNamespace)
+
+	if c.TaskData.ForbiddenNamespace != "" {
+		c.TaskData.TargetNamespace = string(c.TaskData.ForbiddenNamespace)
+	} else {
+		c.TaskData.TargetNamespace = options.ResolveNamespace(c.TaskData.TargetTemplateNamespace, options.TestNamespace)
+	}
+
+	c.TaskData.Template = testtemplate.NewCirrosServerTinyTemplate().Build()
```

Comment:
This method seems too hard coded and hard to configure. Can we just pass these values (such as name, template) to the taskData to also represent what the test case tests?

I know it is hard to understand to begin with, but that is due to a need of setting the right namespaces (since they are generated).